aclk/aclk_tx_msgs.c

```diff
@@ -32,6 +32,36 @@ static void aclk_send_message_subtopic(mqtt_wss_client client, json_object *msg,
 #endif
 }
 
+static uint16_t aclk_send_bin_message_subtopic_pid(mqtt_wss_client client, char *msg, size_t msg_len, enum aclk_topics subtopic, const char *msgname)
+{
+#ifndef ACLK_LOG_CONVERSATION_DIR
+    UNUSED(msgname);
+#endif
+    uint16_t packet_id;
+    const char *topic = aclk_get_topic(subtopic);
+
+    if (unlikely(!topic)) {
+        error("Couldn't get topic. Aborting mesage send");
+        return 0;
+    }
+
+    mqtt_wss_publish_pid(client, topic, msg, msg_len,  MQTT_WSS_PUB_QOS1, &packet_id);
+#ifdef NETDATA_INTERNAL_CHECKS
+    aclk_stats_msg_published(packet_id);
+#endif
+#ifdef ACLK_LOG_CONVERSATION_DIR
+#define FN_MAX_LEN 1024
+    char filename[FN_MAX_LEN];
+    snprintf(filename, FN_MAX_LEN, ACLK_LOG_CONVERSATION_DIR "/%010d-tx-%s.bin", ACLK_GET_CONV_LOG_NEXT(), msgname);
+    FILE *fptr;
+    fptr = fopen(filename,"w");
```

Comment:
Please, add a test to check that you did not received `NULL` as result. Another thing, considering that you are using here `fopen(3)`, `fwrite(3)`, and `fclose(3)`, instead to call the syscalls `open(2)`, `write(2)`, and `close(2)`. I will give a suggestion for you to use the same functions in other file, and of course, this means that we can create an unique function for both.

Reply:
as pre previous comments please ignore all under #ifdef ACLK_LOG_CONVERSATION_DIR for now
it is meant to be only for developer use during development/debugging and will be reworked/improved in future PRs